hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -25,8 +25,7 @@
 
 import hypothesis.internal.conjecture.utils as cu
 import hypothesis.strategies._internal.core as st
-from hypothesis import Verbosity, assume
-from hypothesis._settings import note_deprecation
+from hypothesis import assume
 from hypothesis.errors import InvalidArgument
 from hypothesis.internal.compat import (
     PY2,
```

Comment:
is this guy ( and the switch on line 45 ) still needed?

Reply:
Probably we should remove the `PY2/PY3` constants from `compat` entirely so we can catch things like this.

Reply:
Definitely not needed - as described in #2218, I've just left that for a second PR to be opened on January 1.  The motivation is purely diff size, because if I did everything at once it would be impossible to review.

If we do want to do everything at once I could target this at a new branch, and we could merge several PRs before eventually releasing them all at once? Personally I prefer the incremental approach!